pinot-server/src/main/java/org/apache/pinot/server/starter/helix/HelixServerStarter.java

```diff
@@ -186,6 +194,16 @@ public HelixServerStarter(String helixClusterName, String zkServer, Configuratio
     serverMetrics.addCallbackGauge("memory.allocationFailureCount", PinotDataBuffer::getAllocationFailureCount);
   }
 
+  private void overwriteServerHostInfo() {
```

Comment:
Can you combine this with `addInstanceTagIfNeeded()` as both of them need to access the `InstanceConfig` to reduce the overhead?